primitives/src/lib.rs

```diff
@@ -101,12 +101,34 @@ pub enum DataProviderId {
     Aggregated = 0,
 }
 
-#[derive(Encode, Decode, Debug, Default, Copy, Clone, PartialEq, Eq, PartialOrd, Ord)]
+#[derive(Encode, Decode, Debug, Default, Copy, Clone, PartialEq, Eq)]
 #[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
 pub struct PriceWithDecimal {
     pub price: Price,
     pub decimal: u8,
 }
+impl Ord for PriceWithDecimal {
+    fn cmp(&self, other: &Self) -> Ordering {
+        if let Some((d, e)) = 10u128
```

Comment:
maybe we can give better names than `d` and `e`

Reply:
fixed